File: Study/Programming Skills/02. Find the Difference/TypeScript/main.js
function findTheDifference(s, t) {
    var result = '';
    var sArr = s.split('');
    var tArr = t.split('');
    sArr.sort();
    tArr.sort();
    for (var i = 0; i < sArr.length; i++) {
        if (sArr[i] !== tArr[i]) {
            result = tArr[i];
            break;
        }
    }
    if (result === '') {
        result = tArr[tArr.length - 1];
    }
    return result;
}
;
console.log(findTheDifference('abcd', 'abcde')); // e
console.log(findTheDifference('', 'y')); // y
console.log(findTheDifference('a', 'aa')); // a

File: Study/Programming Skills/02. Find the Difference/GO/main.go
package main

import "fmt"

func main() {
  fmt.Println(findTheDifference("abcd", "abcde"))
  fmt.Println(findTheDifference("", "y"))
  fmt.Println(findTheDifference("a", "aa"))
  fmt.Println(findTheDifference("ae", "aea"))
}


func findTheDifference(s string, t string) byte {
  var result byte
  for i := 0; i < len(s); i++ {
    result ^= s[i] // XOR
  } // XOR of all elements in s and t will be the extra element in t
  for i := 0; i < len(t); i++ {
    result ^= t[i] // XOR
  } // XOR of all elements in s and t will be the extra element in t

  return result
}
